src/ipc/ipc-zephyr.c

```diff
@@ -76,6 +77,28 @@ static bool message_handler(const struct device *dev, void *arg, uint32_t data,
 	return false;
 }
 
+static bool done_handler(const struct device *dev, void *arg)
+{
+	ARG_UNUSED(dev);
+	struct ipc *ipc = (struct ipc *)arg;
+
+	/* task mask not set - nothing to do */
+	if (ipc->task_mask == 0)
+		return false;
```

Comment:
now this whole check isn't needed - this case would be caught by the next `if` too

Reply:
In this implementation, done_handler is invoked every time IPC message is processed.
We can safely assume that most of the IPC messages are not related to power gating.
It is performance wise better to instantly check for most common case and return from function (task_mask is 0).

Reply:
mmh, you really think a check `(x == 0)` is so much faster than `((x & z) == 0)` when calculated once per IPC message, it's like a single asm instruction?

Reply:
not really in this particular case - until we have only 1 use case (entering D3 state and single task mask).
As soon as we add here D0 or some other power state, it will make more and more sense.
What I'm getting at is that this algorithm is slower:

1. Check task mask for power down
4. Check another task mask? Like IPC_TASK_SECONDARY_CORE?
5. Nothing to do, return

Than:
1. Check task mask for 0
2. Nothing to do - return.

Besides, for developers not very familiar with this code we give simple stupid information that this handler does nothing.
I like to keep things simple stupid.

Reply:
Dropped implementation of done_handler.